packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -331,6 +337,46 @@ macro_rules! impl_tx_builder_trait {
                     .any(|input| matches!(input, Input::ResourcePredicate { .. }))
             }
 
+            fn intercept_burn(&self) -> Result<()> {
+                use std::collections::HashSet;
+
+                if self.enable_burn {
+                    return Ok(());
+                }
+
+                let assets_w_change = self
+                    .outputs
+                    .iter()
+                    .filter_map(|output| match output {
+                        Output::Change { asset_id, .. } => Some(*asset_id),
+                        _ => None,
+                    })
+                    .collect::<HashSet<_>>();
+
+                let input_assets = self
+                    .inputs
+                    .iter()
+                    .filter_map(|input| match input {
+                        Input::ResourceSigned { resource } => resource.coin_asset_id(),
+                        Input::ResourcePredicate { resource, .. } => resource.coin_asset_id(),
+                        _ => None,
```

Comment:
What about messages , how do we protect against burning them?

Reply:
Messages should now be properly covered as well